proposals/dictionary-expressions.md

```diff
@@ -33,13 +33,20 @@ An inclusive solution is needed for C#. It should meet the vast majority of case
 The following grammar productions are added:
 
 ```diff
++ collection_expression
+  : '[' ']'
+- | '[' collection_element ( ',' collection_element )* ']'  
++ | '[' 'comparer': expression ']'
+  | '[' ('comparer': expression , )? collection_element ( ',' collection_element )* ']'
+  ;
+
 collection_element
   : expression_element
   | spread_element
-+ | dictionary_element  
++ | key_value_pair_element
   ;
 
-+ dictionary_element
++ key_value_pair_element
 +  : expression ':' expression
```

Comment:
renamed this since its not exclusive to creating dictionaries.

Reply:
we could also consider `pair_element`.  but i don't necessarily mind `key_value_pair_element` as we will allow it to convert to `KeyValuePair<,>` in the case of a non-dictionary collection.

Reply:
I like `key_value_pair_element`. As a fallback, perhaps `associative_element`.